### PR TITLE
Add lexer for Promela

### PR DESCRIPTION
**Pull request by lmarzen**
This pull request adds a lexer for Promela. (https://spinroot.com/spin/whatispin.html)

Promela (**Pro**cess **Me**ta **La**nguage) is a C-like language used by SPIN to formally model software and hardware systems. SPIN (**Si**mple **Pr**omela **I**nterpreter) was originally developed in the 1980s at Bell Labs and since 1991 has been open-source and is a popular formal methods tool.

Pygments already supports other popular formal methods languages, including Isabelle, Coq, and NuSmv. Promela would be a fitting addition.

~~I have not included test snippets since PromelaLexer is derived from CLexer, though I have tested that it produces correct highlighting on a few of my own Promela files.~~ Edit: I added snippets and examplefiles and ran `tox -e check` and `tox -e mapfiles` which both succeeded.

### CODE REVIEW

**Comment by cmarqu** [comment]
(Another potential pygments contributor here...) 
You might want to add a line to the `CHANGES` file also.

**Comment by jeanas** [comment]
> You might want to add a line to the CHANGES file also.

Well, we (maintainers) update it ourselves; we don't ask contributors to do this. (But thank you for taking a look at the PR.)

File: pygments/lexers/c_like.py [hunk 1]
```diff
@@ -657,3 +657,81 @@ class OmgIdlLexer(CLexer):
             include('annotation_appl'),
         ],
     }
+
+
+class PromelaLexer(CLexer):
+    """
+    For the Promela language used with SPIN.
+    """
```
Comment (jeanas) [inline pygments/lexers/c_like.py:665]:
```suggestion
    """
    
```

File: pygments/lexers/c_like.py [hunk 2]
```diff
@@ -657,3 +657,82 @@ class OmgIdlLexer(CLexer):
...
+    
+    name = 'Promela'
+    aliases = ['promela']
+    filenames = ['*.pml', '*.prom', '*.prm', '*.promela', '*pr', '*.pm']
```
Comment (jeanas) [inline pygments/lexers/c_like.py:669]: Shouldn't this be

```suggestion
    filenames = ['*.pml', '*.prom', '*.prm', '*.promela', '*.pr', '*.pm']
```

?

Comment (lmarzen) [reply pygments/lexers/c_like.py]: Good catch. That was a typo.

File: pygments/lexers/c_like.py [hunk 3]
```diff
@@ -657,3 +657,82 @@ class OmgIdlLexer(CLexer):
...
+            # Declarators (suffixes)
+            (words((
+                'priority', 'provided'), suffix=r'\b'),
+             Keyword),
+            # MetaTerms (declarators)
+            (words((
+                'inline', 'ltl', 'select'), suffix=r'\b'),
+             Keyword.Declaration),
+            # MetaTerms (keywords)
+            (r'(skip)\b', Keyword),
```
Comment (jeanas) [inline pygments/lexers/c_like.py:736]:
```suggestion
            (r'skip\b', Keyword),
```

File: pygments/lexers/_mapping.py [hunk 4]
```diff
@@ -391,6 +391,7 @@
     'ProcfileLexer': ('pygments.lexers.procfile', 'Procfile', ('procfile',), ('Procfile',), ()),
     'PrologLexer': ('pygments.lexers.prolog', 'Prolog', ('prolog',), ('*.ecl', '*.prolog', '*.pro', '*.pl'), ('text/x-prolog',)),
     'PromQLLexer': ('pygments.lexers.promql', 'PromQL', ('promql',), ('*.promql',), ()),
+    'PromelaLexer': ('pygments.lexers.c_like', 'Promela', ('promela',), ('*.pml', '*.prom', '*.prm', '*.promela', '*pr', '*.pm'), ('text/x-promela',)),
```
Comment (jeanas) [inline pygments/lexers/_mapping.py:394]:
```suggestion
    'PromelaLexer': ('pygments.lexers.c_like', 'Promela', ('promela',), ('*.pml', '*.prom', '*.prm', '*.promela', '*.pr', '*.pm'), ('text/x-promela',)),
```

**Review by jeanas — COMMENTED** [review]
Looks OK. Since it seems this language can contain embedded C, shouldn't `'types'` and `'keywords'` have an `inherit` at the end to also highlight C types/keywords?

**Comment by lmarzen** [comment]
Promela does not have all the types and keywords of C. Only the ones I specified, so `inherit`, was initentially left out.

File: pygments/lexers/c_like.py [hunk 5]
```diff
@@ -657,3 +657,82 @@ class OmgIdlLexer(CLexer):
...
+
+    # Promela's language reference:
+    # https://spinroot.com/spin/Man/promela.html
+    # Promela's grammar definition:
+    # https://spinroot.com/spin/Man/grammar.html
+
+    tokens = {
+        'statements': [
+            (r'(\[\]|<>|/\\|\\/)|(U|W|V)\b', Operator), # LTL Operators
+            (r'[@]', Punctuation), #remoterefs
```
Comment (jeanas) [inline pygments/lexers/c_like.py:682]:
```suggestion
            (r'@', Punctuation), #remoterefs
```

File: pygments/lexers/c_like.py [hunk 6]
```diff
@@ -657,3 +657,82 @@ class OmgIdlLexer(CLexer):
...
+                '_', '_last', '_nr_pr', '_pid', '_priority', 'else', 'np_',
+                'STDIN'), suffix=r'\b'),
+             Name.Builtin),
+            # Predefined (functions)
+            (words((
+                'empty', 'enabled', 'eval', 'full', 'len', 'nempty', 'nfull',
+                'pc_value'), suffix=r'\b'),
+             Name.Function),
+            # Predefined (operators)
+            (r'(run)\b', Operator.Word),
```
Comment (jeanas) [inline pygments/lexers/c_like.py:720]:
```suggestion
            (r'run\b', Operator.Word),
```

**Comment by jeanas** [comment]
Sorry for the delay. Thanks for your contribution!

**Pull request merged**